src/lib/dlt_user.c

```diff
@@ -2917,9 +2917,20 @@ void dlt_user_trace_network_segmented_thread(void *unused)
     prctl(PR_SET_NAME, "dlt_segmented", 0, 0, 0);
 #endif
 
+    int count = 0;
     s_segmented_data *data;
 
     while (1) {
+        // Normally reattaching to dlt-daemon is triggered when app tries to output log.
+        // Even if the app outputs log only before starting up of the dlt-daemon,
+        // this retry loop helps reattaching.
+        while ((dlt_user.dlt_log_handle < 0) && (count < DLT_USER_REATTACH_MAX_COUNT))
```

Comment:
This condition already checked in dlt_user_log_reattach_to_daemon(). 
As my opinion, you do not need to retry loop here, just do the same as dlt_user_receiverthread_function. That mean just call the dlt_user_log_reattach_to_daemon() before dlt_user_trace_network_segmented_thread_segmenter(data).

Reply:
As ssugiura also mentioned, we also need to flush normal log here.
If application does not send any network traces, network segmented thread is blocked on pthread_cond_wait() in line 2938. Therefore reattaching to daemon is tried before blocking point.

Reply:
@ysk-sato can you, please, provide the code for that?

Reply:
I meant that current source code already realized that. (= Already segmented thread will flush normal log). I thought that still we need retry loop here and we don't have to change code here.